pkg/log/log_file_test.go

```diff
@@ -0,0 +1,190 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main under directory cmd parses and validates user input,
+// instantiates and initializes objects imported from pkg, and runs
+// the process.
+package log
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/component-base/logs"
+	"k8s.io/klog"
+)
+
+var (
+	testFlags          = initFlags()
+	klogDefaultMaxSize = klog.MaxSize
+)
+
+func initFlags() *pflag.FlagSet {
+	flags := pflag.NewFlagSet("test", pflag.ContinueOnError)
+	AddFlags(flags)
+	flags.AddGoFlagSet(flag.CommandLine)
+	return flags
+}
+
+func restoreFlagDefaultValues() {
+	testFlags.Set(logToStdErrFlag, "true")
+	testFlags.Set(logFileFlag, "")
+	testFlags.Set(logDirFlag, "")
+	testFlags.Set(maxSizeFlag, fmt.Sprintf("%d", klogDefaultMaxSize/1024/1024))
+	testFlags.Set(maxNumFlag, "0")
+
+	klog.MaxSize = klogDefaultMaxSize
+	logFileMaxNum = 0
+	logDir = ""
+}
+
+// Log ~100K bytes.
+func testLogging() {
+	// ~1K bytes per line.
+	line := make([]byte, 1000)
+	for i := 0; i < 1000; i++ {
+		line[i] = byte('0' + i%10)
+	}
+	// Log 100 lines, ~100K bytes.
+	for i := 0; i < 100; i++ {
+		klog.Infof("%d: %s", i, string(line))
+		klog.Warningf("%d: %s", i, string(line))
+	}
+	logs.FlushLogs()
+}
+
+func TestKlogFileLimits(t *testing.T) {
+	testLogDir, err := ioutil.TempDir("", "antrea-log-test")
+	if err != nil {
+		t.Errorf("Failed to create tmp log dir: %v", err)
+		return
+	}
+	defer os.RemoveAll(testLogDir)
+
+	args := []string{"--logtostderr=false", "--log_dir=" + testLogDir, "--log_file_max_size=1", "--log_file_max_num=2"}
+	testFlags.Parse(args)
+	InitLogFileLimits(testFlags)
+	logs.InitLogs()
+	defer restoreFlagDefaultValues()
+
+	// Should generate 3 log files (100K * 30 / 1M).
+	for i := 0; i < 30; i++ {
+		testLogging()
+		time.Sleep(time.Millisecond * 50)
+	}
+
+	infoLogFileNum := 0
+	warningLogFileNum := 0
+	validateFn := func() {
+		f, err := os.Open(testLogDir)
+		if err != nil {
+			t.Errorf("Failed to open log directory: %v", err)
+			return
+		}
+		allFiles, err := f.Readdir(-1)
+		if err != nil {
+			t.Errorf("Failed to read log directory: %v", err)
+			return
+		}
+		f.Close()
+
+		infoLogFiles := []os.FileInfo{}
+		warningLogFiles := []os.FileInfo{}
+		for _, file := range allFiles {
+			if !file.Mode().IsRegular() {
+				// Skip dir, symbol link, etc.
+				continue
+			}
+			if !strings.HasPrefix(file.Name(), executableName) {
+				continue
+			}
+			if strings.Contains(file.Name(), ".log.INFO.") {
+				infoLogFiles = append(infoLogFiles, file)
+			} else if strings.Contains(file.Name(), ".log.WARNING.") {
+				warningLogFiles = append(warningLogFiles, file)
+			} else if strings.Contains(file.Name(), ".log.ERROR.") {
+			} else {
```

Comment:
would be good to have a comment here stating that the test is purposely not generating any error logs, or remove the empty if statement altogether

Reply:
Added comments.